experimental/InvariantTheory/src/InvariantTheory.jl

```diff
@@ -412,17 +440,27 @@ function Base.show(io::IO, R::RedGroupInvarRing)
     io = pretty(io)
     println(io, "Invariant Ring of")
     println(io, Lowercase(), R.poly_ring)
-    print(io, Indent(),  "under group action of ", R.group.group[1], R.group.group[2])
-    print(io, Dedent())
+    if isdefined(group(R), :group)
+        print(io, Indent(),  "under group action of ", R.group.group[1], R.group.group[2])
+        print(io, Dedent())
+    else
+        println(io, Indent(),  "under group action of linearly reductive group defined by")
+        print(io, group_ideal(group(R)))
+        print(io, Dedent())
+    end
 end
 
 #computing the graph Gamma from Derksens paper
 function image_ideal(G::LinearlyReductiveGroup, rep_mat::MatElem)
     R = base_ring(rep_mat)
     n = ncols(rep_mat)
-    m = G.group[2]
-    mixed_ring_xy, x, y, zz = polynomial_ring(G.field, "x"=>1:n, "y"=>1:n, "zz"=>1:m^2)
-    ztozz = hom(R,mixed_ring_xy, gens(mixed_ring_xy)[(2*n)+1:(2*n)+(m^2)])
+    if isdefined(G, :group)
+        m = (G.group[2])^2
+    else
+        m = ngens(base_ring(group_ideal(G)))
+    end
+    mixed_ring_xy, x, y, zz = polynomial_ring(G.field, "x"=>1:n, "y"=>1:n, "zz"=>1:m)
+    ztozz = hom(R,mixed_ring_xy, gens(mixed_ring_xy)[(2*n)+1:(2*n)+(m)])
```

Comment:
Does this also work?
```suggestion
    ztozz = hom(R, mixed_ring_xy, zz)
```